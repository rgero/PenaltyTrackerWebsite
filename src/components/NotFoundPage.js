import React from 'react';
import { Link } from 'react-router-dom';
import Header from './Header';

const NotFoundPage = () => (
  <div>
    <Header/>

    <div className="content-container">
      <div className="content-container__title">
        Page Not Found
      </div>
      <div>
      The page you are trying to access does not exist, please return <Link to="/">home</Link>
      </div>
    </div>
  </div>
);

export default NotFoundPage;
