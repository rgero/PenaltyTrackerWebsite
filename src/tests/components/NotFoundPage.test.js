import React from 'react';
import { shallow } from 'enzyme';
import NotFoundPage from '../../components/NotFoundPage';

test("Testing Not Found rendering", ()=>{
    const wrapper = shallow(<NotFoundPage />);
    expect(wrapper).toMatchSnapshot();
})