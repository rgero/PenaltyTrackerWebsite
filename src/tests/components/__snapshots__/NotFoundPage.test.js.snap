// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Not Found rendering 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFoundPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <div
          className="content-container"
        >
          <div
            className="content-container__title"
          >
            Page Not Found
          </div>
          <div>
            The page you are trying to access does not exist, please return 
            <Link
              replace={false}
              to="/"
            >
              home
            </Link>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="content-container__title"
            >
              Page Not Found
            </div>,
            <div>
              The page you are trying to access does not exist, please return 
              <Link
                replace={false}
                to="/"
              >
                home
              </Link>
            </div>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Page Not Found",
              "className": "content-container__title",
            },
            "ref": null,
            "rendered": "Page Not Found",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "The page you are trying to access does not exist, please return ",
                <Link
                  replace={false}
                  to="/"
                >
                  home
                </Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "The page you are trying to access does not exist, please return ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "home",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "home",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <div
            className="content-container"
          >
            <div
              className="content-container__title"
            >
              Page Not Found
            </div>
            <div>
              The page you are trying to access does not exist, please return 
              <Link
                replace={false}
                to="/"
              >
                home
              </Link>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="content-container__title"
              >
                Page Not Found
              </div>,
              <div>
                The page you are trying to access does not exist, please return 
                <Link
                  replace={false}
                  to="/"
                >
                  home
                </Link>
              </div>,
            ],
            "className": "content-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Page Not Found",
                "className": "content-container__title",
              },
              "ref": null,
              "rendered": "Page Not Found",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "The page you are trying to access does not exist, please return ",
                  <Link
                    replace={false}
                    to="/"
                  >
                    home
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "The page you are trying to access does not exist, please return ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "home",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "home",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
